modules/core/migrations/v7/genesis.go

```diff
@@ -0,0 +1,40 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	clientv7 "github.com/cosmos/ibc-go/v6/modules/core/02-client/migrations/v7"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/types"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove any localhost clients
+func MigrateGenesis(appState genutiltypes.AppMap, cdc codec.ProtoCodecMarshaler) (genutiltypes.AppMap, error) {
+	if appState[host.ModuleName] != nil {
+		// ensure legacy solo machines are registered
+		clientv7.RegisterInterfaces(cdc.InterfaceRegistry())
```

Comment:
will `RegisterInterfaces` panic if the interface registry is already registered? I remember seeing some behaviour where duplicate interface registration panicked.

Reply:
We don't want the user to call `RegisterInterfaces` before hand. The migration registration should not persist onto the chain codec, it should only be used to perform the genesis migrations. I can see if I can add a test to ensure it panics

Reply:
The codec will not panic if the types are already registered. It only panics if you try to register duplicate types (same URL, different generated files)

Reply:
great that makes sense thanks